apps/dcos_rest/src/dcos_rest_vips_handler.erl

```diff
@@ -42,7 +42,12 @@ vip({{Protocol, VIP, Port}, riak_dt_orswot}) ->
     vip({Protocol, VIP, Port});
 vip({Protocol, {name, {Id, Framework}}, Port}) ->
     [ZoneName|_ZoneNames] = ?ZONE_NAMES,
-    FullName = mesos_state:domain_frag([Id, Framework | ZoneName]),
+    %% mesos_state:domain_frag/1 replaces dots with dashes in given fragments.
+    %% If Id contains a dot, it is going to be replaced wit a dash too.
+    %% Let's prevent it from happening by splitting Id using dots, and let
+    %% mesos_state:domain_frag/1 join them back with dots.
+    IdFragments = binary:split(Id, <<".">>, [global, trim_all]),
+    FullName = mesos_state:domain_frag(IdFragments ++ [Framework | ZoneName]),
```

Comment:
```
1> mesos_state:domain_frag([<<"/http.server">>]).
<<"http-server">>
2> mesos_state:domain_frag(<<"/http.server">>).  
<<"http.server">>
```

It's ludicrous, but `mesos_state:domain_frag/1` returns `-` if argument is list and `.` if argument is binary.

Could you please export `dcos_l4lb_lashup_vip_listener:to_name/1` and use it here?

https://github.com/dcos/dcos-net/blob/f3f6820/apps/dcos_l4lb/src/dcos_l4lb_lashup_vip_listener.erl#L313-L317